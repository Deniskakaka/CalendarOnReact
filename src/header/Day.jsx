import React from "react";
import moment from "moment";
import './day.scss';

const Day = props => {
  const today = moment().format("YYYY-MM-D");
  return props.ArrayOFWeek.map(day => (
    <div className="day" key={day}>
      <span className="day__name">
        {moment()
          .day(day)
          .format("ddd")}
      </span>
      <span
        className={
          today ===
          moment()
            .day(day)
            .format("YYYY-MM-D")
            ? "day__number today"
            : "day__number"
        }
      >
        {moment()
          .day(day)
          .format("D")}
      </span>
    </div>
  ));
};
export default Day;
