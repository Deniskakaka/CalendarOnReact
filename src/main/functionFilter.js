import moment from "moment";

export const filterTasks = (date, array) => {
  let mass = array.filter(i => i.timeStart === date);
  return mass;
};

export const top = time => {
  const hours = +time.slice(0, 2) * 61;
  const minuts = +time.slice(3);
  return hours + minuts;
};

export const height = (timeStart, timeEnd) => {
  const hours =
    +timeStart.slice(0, 2) === 23 &&
    +timeEnd.slice(0, 2) === 0
      ? 61
      : Math.abs(
          +timeStart.slice(0, 2) - +timeEnd.slice(0, 2)
        ) * 61;
  const minuts =
    +timeStart.slice(0, 2) === 23 &&
    +timeEnd.slice(0, 2) === 0
      ? 0
      : Math.abs(+timeStart.slice(3) - timeEnd.slice(3));
  return hours + minuts;
};

export const getRandomIntInclusive = (min, max) => {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min + 1)) + min;
};

export const sameTime = (tasks, object) => {
  let sameTime = [];
  for (let i = 0; i < tasks.length; i++) {
    if (
      tasks[i].timeStart.slice(5, 7) ===
        object.timeStart.slice(5, 7) &&
      tasks[i].timeStart.slice(8) ===
        object.timeStart.slice(8) &&
      +tasks[i].start.slice(0, 2) ===
        +object.start.slice(0, 2) &&
      +tasks[i].start.slice(3) === +object.start.slice(3)
    ) {
      sameTime.push(object);
    }
    if (
      tasks[i].timeStart.slice(5, 7) ===
        object.timeStart.slice(5, 7) &&
      tasks[i].timeStart.slice(8) ===
        object.timeStart.slice(8) &&
      +tasks[i].start.slice(0, 2) <
        +object.start.slice(0, 2) &&
      +tasks[i].end.slice(0, 2) > +object.start.slice(0, 2)
    ) {
      sameTime.push(object);
    }
    if (
      tasks[i].timeStart.slice(5, 7) ===
        object.timeStart.slice(5, 7) &&
      tasks[i].timeStart.slice(8) ===
        object.timeStart.slice(8) &&
      +tasks[i].start.slice(0, 2) ===
        +object.start.slice(0, 2) &&
      +tasks[i].start.slice(3) < object.start.slice(3)
    ) {
      sameTime.push(object);
    }
  }
  return sameTime;
};

export const overDay = object => {
  return (object.start.slice(0, 2) === 23 &&
    object.end.slice(0, 2) > 0) ||
    object.timeStart < object.timeEnd
    ? true
    : false;
};

export const pastDay = object => {
  return object.timeStart < moment().format("YYYY-MM-DD") ||
    object.timeEnd < moment().format("YYYY-MM-DD")
    ? true
    : false;
};
