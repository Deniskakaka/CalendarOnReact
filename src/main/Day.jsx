import React from 'react';
import './day.scss';
import Hours from './Hours.jsx';
import moment from 'moment';
import { filterTasks, top, height } from './functionFilter.js';

class Day extends React.Component {

    state = {
        time: top(moment().format('HH:mm'))
    }
    
    componentDidMount() {
        this.interval = setInterval(() => {
            this.setState({
                time: top(moment().format('HH:mm'))
            })
        }, 60000)
    }

    componentWillUnmount() {
        clearInterval(this.interval)
    }

    render() {
        return (
            this.props.ArrayOFWeek.map(day => 
            <div className="dayWeek" key={day} onClick={this.props.openPopap}>
                {moment().format('YYYY-MM-DD') === moment().day(day).format("YYYY-MM-DD") ? <div className="redLine" style={{top:`${+this.state.time}px`}}></div> : null}
                {filterTasks(moment().day(day).format('YYYY-MM-DD'), this.props.tasks).map(i => 
                <div 
                    className="task" 
                    style={{top:`${top(i.start)}px`,height:`${height(i.start, i.end)}px`}} 
                    key={day} 
                    onClick={(e) => this.props.showData(i.start,i.end,i.timeStart,i.timeEnd,i.id)}>
                    <span className="task__title">{i.title}</span>
                    <div>
                        <span className="task__time">{i.start}</span>
                        <span className="task__line">-</span>
                        <span className="task__time">{i.end}</span>
                    </div>
                <span>{i.description}</span>
                </div>)}
                <Hours changeTime={this.props.changeTime} showData={this.props.showData} day={day} showHours={this.props.showHours}/>
            </div>)
        );
    }
    
};

export default Day;