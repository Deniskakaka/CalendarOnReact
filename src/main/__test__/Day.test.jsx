import React from 'react';
import { shallow } from 'enzyme';
import Day from '../Day.jsx';
import { filterTasks, top }  from '../functionFilter.js';

jest.mock('../functionFilter.js', () => {
    return {
        filterTasks: jest.fn(() => {
           return [{
            "id": "1",
            "title": "not task",
            "start": "01:00",
            "end": "02:00",
            "timeStart": "2020-02-10",
            "timeEnd": "2020-02-10",
            "description": ""
        }]}),
        top: jest.fn(() => '21:32'),
        height: jest.fn(() => '10:00', '13:00')
    }
});


describe('<Day/>', () => {
    it('should display show class task', () => {
        const wrappedComponent = shallow(<Day ArrayOFWeek={[1,2,3,4,5,6,7]}/>)
        expect(wrappedComponent.find('.task').exists()).toBeTruthy()
    });

    it('should display show class task', () => {
        const wrappedComponent = shallow(<Day ArrayOFWeek={[1,2,3,4,5,6,7]}/>)
        expect(wrappedComponent.find('Hours')).toMatchSnapshot(true)
    })
});