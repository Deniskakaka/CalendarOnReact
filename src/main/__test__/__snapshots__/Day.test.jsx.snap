// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Day/> should display show class task 1`] = `
Array [
  <Hours
    day={1}
  />,
  <Hours
    day={2}
  />,
  <Hours
    day={3}
  />,
  <Hours
    day={4}
  />,
  <Hours
    day={5}
  />,
  <Hours
    day={6}
  />,
  <Hours
    day={7}
  />,
]
`;
